SimplifiedTests/UserProfileDocumentTests.swift

```diff
@@ -62,4 +198,68 @@ class UserProfileDocumentTests: XCTestCase {
       XCTAssert(false, error.localizedDescription)
     }
   }
+    
+  func testParseJSONInvalid() {
+    let data = dataCorruptedJson.data(using: .utf8)
+    XCTAssertNotNil(data)
+    
+    do {
+      let _ = try UserProfileDocument.fromData(data!)
+      XCTAssert(false)
+    } catch {
+      let err = error as NSError
+      XCTAssertEqual(err.code, UserProfileDocument.decodingErrorCodeInvalid)
+        
+      let customErrorCode = err.userInfo[UserProfileDocument.parseErrorKey] as? Int ?? -1
+      XCTAssertEqual(customErrorCode, UserProfileDocument.NYPLParseProfileErrorKey.dataCorrupted.rawValue)
+    }
+  }
+    
+  func testParseJSONMissingProperty() {
+    let data = keyNotFoundJson.data(using: .utf8)
+    XCTAssertNotNil(data)
+    
+    do {
+      let _ = try UserProfileDocument.fromData(data!)
+      XCTAssert(false)
+    } catch {
+      let err = error as NSError
+      XCTAssertEqual(err.code, UserProfileDocument.decodingErrorCodeNotFound)
+        
+      let customErrorCode = err.userInfo[UserProfileDocument.parseErrorKey] as? Int ?? -1
```

Comment:
is the `-1` indicative of a failure? If so I would prefer to have a guard-let instead and XCTFail inside its else so that it's more explicit.

Reply:
Added guard let instead of -1